bindings/python/cconfigspace/configuration.py

```diff
@@ -44,5 +57,28 @@ def configuration_space(self):
     return self._configuration_space
 
   def check(self):
+    """Check the validity of the current configuration based on the generative configuration space.
+    """
     res = ccs_configuration_check(self.handle)
     Error.check(res)
+
+  @property
+  def hyperparameters(self):
+    """Returns the list of hyperparameters of the current configuration."""
+    return self.configuration_space.hyperparameters
+
+  def __getitem__(self, idx):
+    if type(idx) is int:
+      return self.values[idx]
+    elif type(idx) is str:
+      return self.value(idx)
+    else:
+      raise ValueError("index should be str or int")
+
+  def to_list(self):
+    """Returns a list copy of hyperparameter values of the current configuration."""
+    return self.values[:]
+
+  def to_dict(self):
```

Comment:
Already implemented as asdict in binding.py. I'll let you be judge of the more pythonic option.

Reply:
yes but I noticed that this code was faster